### PR TITLE
Add example of resolving sequential pulses.

### PR DESCRIPTION
**Pull request by molpopgen**
Add detailed description of pulse resolution.
The content comes from posts in #100.

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -374,6 +374,100 @@ See the
 {ref}`sec_spec_mdm_population_sizes` section for more details on
 how proportions should be interpreted.
 
+## Example: sequential application of pulses
```
Comment (grahamgower) [inline docs/specification.md:377]: This heading seems to be at the wrong level. Take a look at the rendered page and you'll see that the nesting in the contents (top right) is off. The `### migration` and `### schema` subsections below this should be under the MDM section.  Maybe it should be 3 or 4 `#` marks instead?

File: docs/specification.md [hunk 2]
```diff
@@ -374,6 +374,100 @@ See the
...
+  time: 10
+- sources: [B]
+  dest: C
+  proportions: [1/4]
+  time: 10
+```
+
+Ten (10) generations ago, pulse events occur from source demes `A` and `B`
+into destination deme `C`.
+We need to arrive at the final ancestry proportions for destination deme `C` at this time.
```
Comment (grahamgower) [inline docs/specification.md:406]: s/at this time/after this time/

File: docs/specification.md [hunk 3]
```diff
@@ -374,6 +374,100 @@ See the
...
+
+The steps are:
```
Comment (grahamgower) [inline docs/specification.md:408]: There are clearly two ways to do the sequential application of pulses: 1) apply each pulse one at a time, like what we can do in msprime, or 2) calculate the resulting proportions as you've indicated and sample the parental generation once.

The wording here strongly suggests that case (2) is the only option.
How about "To arrive at the final ancestry proportions for deme `C` after this time:" instead?

**Review by grahamgower — CHANGES_REQUESTED** [review]
Just a few minor wording things.

There's also a section in the tutorial that talks about simultaneous pulses (click on the "answer" to the question below example 17). Maybe some cross-linking between the tutorial and the spec is in order here?
https://popsim-consortium.github.io/demes-spec-docs/main/tutorial.html#a-pulse-of-admixture

File: docs/specification.md [hunk 4]
```diff
@@ -374,6 +374,100 @@ See the
...
+    dest: C
+    proportions: [1/3]
+    time: 10
+  - sources: [B]
+    dest: C
+    proportions: [1/4]
+    time: 10
+  ```
+
+  We get `[1/3, 1/6, 1/2]` as our ancestry proportions due to pulses.
```
Comment (grahamgower) [inline docs/specification.md:435]: Hang on, isn't this is the same model as above? How are the ancestry proportions different?

File: docs/specification.md (hunk 3 above)
Comment (molpopgen) [reply docs/specification.md:408]: This comment is the heart of the issue: how is it clear if I'm having to  ask/open issue about what you mean by "apply"?  How would I be expected to know what msprime does?

Comment (grahamgower) [reply docs/specification.md:408]: Don't get me wrong here, I agree with you 100% that additional clarification is needed here. But I think the wording should be more agnostic to how one would implement this, and instead frame the explanation in terms of what the realised ancestry proportions must be after the pulse time.

Comment (molpopgen) [reply docs/specification.md:408]: If you think there is more than one way and all of them work for discrete and continuous simulators, then let me know.  I've never looked at the msprime code, and I probably won't.  I only think about doing this stuff forwards in time.  For that case, "applying things one at a time" doesn't mean there is only one implementation unless you specify exactly how to apply it.  Someone could take the first pulse in this example, draw p*N children from the source deme and (1-p)*N from the destination deme.  Then, the remaining pulses do nothing as the code would probably realize there are no further children needed.  The notion that that is a wrong thing to do is in the comments of #100, but I'd have a hard time arguing they misread the spec--they did indeed apply pulses one at a time.

Comment (molpopgen) [reply docs/specification.md:408]: I think part of the issue may be that we are trying to describe this in words rather than with an equation.

Comment (grahamgower) [reply docs/specification.md:408]: In my head, sequential application means: For the first pulse into destination `C`, draw `p*N` individuals from source deme `A` and `(1-p)*N` from destination deme `C`. For the second pulse into dest `C`, draw `q*N` individuals from the source deme `B` and `(1-q)*N` from the destination deme `C`, where the destination deme `C` is currently made up of `p*N` individuals from `A` and `(1-p)*N` individuals from `C`.

For an individual-based simulator it makes sense to code up this procedure to calculate the realised ancestry proportions and just sample those realised proportions directly. For a coalescent simulator, there is no need to do this as one can just "apply" the pulses in reverse order (i.e. backwards time): observe the second pulse event and move lineages with probability `q`, then observe the first pulse event and move lineages with probability `p`.

Comment (molpopgen) [reply docs/specification.md:408]: I think the trick is to write this down for any number of sources :)

Comment (grahamgower) [reply docs/specification.md:408]: Here you go :P
https://github.com/tskit-dev/msprime/blob/4937d882fe6239d4dcdbe3c94916ce47b380c082/msprime/demography.py#L3511-L3537

**Comment by molpopgen** [comment]
Little delay on this one: the example is bad.  It has `1/3` in the YAML, but the demes-python rejects that (string not a number).  I'll try to come up with an example that actually works.  This specific example gets confounded w/numerical issues due to trying to write `1/3` as a number, which is actually a whole other concern I hadn't thought of.

File: docs/specification.md [hunk 5]
```diff
@@ -374,6 +374,117 @@ See the
...
+For the above model, the steps are:
+
+1. `x = [0, 0, 0]`
+2. `x = [0, 0, 1]`
+3. `p = 1 - 0.25`
+   x = `x*p` = [0, 0, 0.75]`
+   `x[A] += 0.25, x = [0.25, 0, 0.75]`
+   `p = 1 - 0.2`
+   `x = x*p = [0.2, 0, 0.6]`
+   `x[B] += 0.2, x = [0.2, 0.2, 0.6]`
```
Comment (grahamgower) [inline docs/specification.md:428]: Thanks for the updates @molpopgen.
The formatting just here makes the rendered result hard to read. Maybe a code block would be clearer? (e.g. allowing newlines)

**Review by grahamgower — APPROVED** [review]
Great, thanks @molpopgen! Feel free to squash and merge.

**Pull request merged**